prime-router/src/main/kotlin/azure/ActionHistory.kt

```diff
@@ -188,9 +185,18 @@ class ActionHistory(
             }
         }
         // capture the azure client IP but override with the first forwarded for if present
-        action.senderIp = request.headers["x-azure-clientip"]?.take(ACTION.SENDER_IP.dataType.length())
+        action.senderIp = if (ACTION.SENDER_IP.dataType.length() > 0) {
```

Comment:
This and the handling of several other strings needed to change in order to not break when the database column has no length limit. `.length()` returns 0 in this case, which caused the string to be truncated to 0 characters.